common/src/main/java/org/opensearch/ml/common/utils/MLTaskUtils.java

```diff
@@ -80,4 +83,22 @@ public static void updateMLTaskDirectly(
             listener.onFailure(e);
         }
     }
+
+    /**
+     * Checks if a given ML task is marked for cancellation by querying its current state.
+     * This method verifies if the task's state is set to CANCELLING.
+     *
+     * @param taskId The ID of the ML task to check
+     * @param client The OpenSearch client to use for querying the task state
+     * @return true if the task exists and is in CANCELLING state, false otherwise
+     */
+    public static boolean isTaskMarkedForCancel(String taskId, Client client) {
+        if (taskId != null && !taskId.isEmpty()) {
+            MLTaskGetRequest taskGetRequest = MLTaskGetRequest.builder().taskId(taskId).build();
+            MLTaskGetResponse taskResponse = client.execute(MLTaskGetAction.INSTANCE, taskGetRequest).actionGet();
```

Comment:
This will block thread, we see several times that such blocking code will cause problems. Suggest use action listener.